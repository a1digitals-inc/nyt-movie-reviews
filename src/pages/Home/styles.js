import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Container = styled.div`
  max-width: 1020px;
  margin: 0 auto;
  padding: 0 20px 50px;

  display: flex;
  flex-direction: column;

  ul {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 15px;
    margin-top: 30px;
  }
`;

export const Option = styled(Link)`
  padding: 20px;
  border-radius: 4px;
  color: #000000;
  background: #fff;
  text-align: center;

  &:hover {
    color: #000000;
    background: #e6e6e6;
  }

  strong {
    display: block;
    font-size: 20px;
    font-weight: normal;
  }

  img {
    height: 60px;
    margin-top: 15px;
  }

  span {
    display: block;
    margin-top: 3px;
    color: #999;
    margin-top: 30px;
  }
`;
