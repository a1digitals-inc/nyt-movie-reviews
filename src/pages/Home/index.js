import React from 'react';

import reviewLogo from '../../assets/images/review.svg';
import criticLogo from '../../assets/images/critic.svg';
import favoriteLogo from '../../assets/images/favorite.svg';
import { Container, Option } from './styles';

export default function Home() {
  return (
    <Container>
      <ul>
        <Option to="/reviews">
          <strong>Reviews</strong>
          <img src={reviewLogo} alt="ReviewLogo" />
          <span>
            Search New York Times movie reviews by keyword and filter by
            Criticâ€™s Pick and Reviewer. Supports ordering results by-title,
            by-publication-date, or by-opening-date.
          </span>
        </Option>

        <Option to="/critics">
          <strong>Critics</strong>
          <img src={criticLogo} alt="CriticLogo" />
          <span>
            Get New York Times movie critics. You can either specify the
            reviewer name.
          </span>
        </Option>

        <Option to="/favorites">
          <strong>Favorites</strong>
          <img src={favoriteLogo} alt="favoriteLogo" />
          <span>Favorites New York Times movie reviews.</span>
        </Option>
      </ul>
    </Container>
  );
}
