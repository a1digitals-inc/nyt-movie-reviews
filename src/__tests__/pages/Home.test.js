import React from 'react';
import { shallow } from 'enzyme';

import Home from '../../pages/Home';

describe('Home page', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
