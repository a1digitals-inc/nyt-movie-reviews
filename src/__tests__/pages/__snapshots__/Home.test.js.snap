// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render without crashing 1`] = `
<styled.div>
  <ul>
    <Styled(Link)
      to="/reviews"
    >
      <strong>
        Reviews
      </strong>
      <img
        alt="ReviewLogo"
        src="review.svg"
      />
      <span>
        Search New York Times movie reviews by keyword and filter by Criticâ€™s Pick and Reviewer. Supports ordering results by-title, by-publication-date, or by-opening-date.
      </span>
    </Styled(Link)>
    <Styled(Link)
      to="/critics"
    >
      <strong>
        Critics
      </strong>
      <img
        alt="CriticLogo"
        src="critic.svg"
      />
      <span>
        Get New York Times movie critics. You can either specify the reviewer name.
      </span>
    </Styled(Link)>
    <Styled(Link)
      to="/favorites"
    >
      <strong>
        Favorites
      </strong>
      <img
        alt="favoriteLogo"
        src="favorite.svg"
      />
      <span>
        Favorites New York Times movie reviews.
      </span>
    </Styled(Link)>
  </ul>
</styled.div>
`;
